baseline_indexer/indexer.cc

```diff
@@ -138,8 +148,13 @@ int main(int argc, char** argv) {
     // the used in indexing array denotes whether a coordinate was used for the
     // fft (might not be if dmin filter was used for example). The used_in_indexing array
     // is sometimes used onwards in the dials indexing algorithms, so keep for now.
+    if (nthreads == 0){ // i.e. user has not specified.
+      nthreads = std::thread::hardware_concurrency(); // Get max number of threads
+    }
+    if (nthreads == 0) nthreads = 1; // Set to default number of threads
+
```

Comment:
You don't need to set a default value in order to get this kind of behaviour, the parser has an `argument.is_used()` function so you could just do:

```suggestion
    size_t nthreads;
    if (parser.is_used("nthreads") {
        nthreads = parser.get<size_t>("nthreads")
    }
    else {
        size_t max_threads = std::thread::hardware_concurrency();
        nthreads = max_threads ? max_threads : 1; 
    }
```

Reply:
Thanks for teaching me about this option, included now.